### PR TITLE
docs: add --run-cache for fetch/pull/push

### PR DESCRIPTION
**Pull request by efiop**
Related to https://github.com/iterative/dvc.org/issues/1289

Per https://github.com/iterative/dvc/pull/4087

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by efiop** [comment]
This is just the declarative portion. Full #1289 solution will be a separate PR.

File: content/docs/command-reference/fetch.md [hunk 1]
```diff
@@ -7,7 +7,9 @@ Get tracked files or directories from
 
 ```usage
 usage: dvc fetch [-h] [-q | -v] [-j <number>]
-                 [-r <name>] [-a] [-T] [-d] [-R] [--all-commits]
+                 [-r <name>] [-a] [-T]
+                 [--all-commits] [-d] [-R]
+                 [--run-cache]
```
Comment (jorgeorpinel) [inline content/docs/command-reference/fetch.md:12]: We can prob. rearrange these into fewer lines (under 72-char each).

File: content/docs/command-reference/fetch.md [hunk 2]
```diff
@@ -77,6 +79,9 @@ or `-T` options are used).
   [remote storage](/doc/command-reference/remote) to fetch from (see
   `dvc remote list`).
 
+- `--run-cache` - downloads all available history of stage runs from the remote
+  repository.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/fetch.md:83]: Is there a way to know how big this is? Or to list all the checkpoints that have been saved between each commit or with timestamps, or something like that? Thanks

File: content/docs/command-reference/pull.md [hunk 3]
```diff
@@ -95,6 +95,9 @@ reflinks or hardlinks to put it in the workspace without copying. See
   [remote storage](/doc/command-reference/remote) to pull from (see
   `dvc remote list`).
 
+- `--run-cache` - downloads all available history of stage runs from the remote
```
Comment (jorgeorpinel) [inline content/docs/command-reference/pull.md:98]: Looks like the command usage in Synopsis wasn't updated in this file 🙂

File: content/docs/command-reference/fetch.md (hunk 2 above)
Comment (efiop) [reply content/docs/command-reference/fetch.md:83]: Nope, it is pretty premitive for now.

**Pull request merged**